front/components/RenderMarkdown.tsx

```diff
@@ -76,14 +141,106 @@ export function RenderMarkdown({ content }: { content: string }) {
         ol: OlBlock,
         li: LiBlock,
         p: ParagraphBlock,
+        sup: CiteBlockWrapper(message),
       }}
-      remarkPlugins={[remarkDirective, customDirectivesPlugin, remarkGfm]}
+      remarkPlugins={[
+        remarkDirective,
+        mentionDirective,
+        citeDirective(),
+        remarkGfm,
+      ]}
     >
       {addClosingBackticks(content)}
     </ReactMarkdown>
   );
 }
 
+function isCiteProps(props: ReactMarkdownProps): props is ReactMarkdownProps & {
+  references: string;
+} {
+  return Object.prototype.hasOwnProperty.call(props, "references");
+}
+
+function CiteBlockWrapper(message: AgentMessageType | UserMessageType) {
+  const CiteBlock = (props: ReactMarkdownProps) => {
+    if (isCiteProps(props) && props.references) {
+      const references = JSON.parse(props.references) as {
+        counter: number;
+        ref: string;
+        link: string;
+      }[];
+
+      const documents: { [key: string]: RetrievalDocumentType } = {};
+      if (isAgentMessageType(message)) {
+        if (
+          message.action &&
+          isRetrievalActionType(message.action) &&
+          message.action.documents
+        ) {
+          message.action.documents.forEach((d) => {
+            documents[d.reference] = d;
+          });
+        }
+      }
+
+      return (
+        <>
+          {references.map((r, i) => {
+            const document = documents[r.ref];
+            const provider = providerFromDocument(document);
+            const title = titleFromDocument(document);
+            const citeClassNames = classNames(
+              "rounded-md bg-structure-100 px-1",
+              "text-xs font-semibold text-action-500",
+              "hover:bg-structure-200 hover:text-action-600"
+            );
+
+            return (
+              <sup key={`${r.ref}-${i}`}>
+                <Tooltip
+                  contentChildren={
+                    <div className="flex flex-row items-center gap-x-1">
```

Comment:
Actually this raises a js error: 
<img width="869" alt="Capture d’écran 2023-09-22 à 13 02 27" src="https://github.com/dust-tt/dust/assets/3803406/348342c4-0ed5-4790-8b90-361fa7404673">